### ISSUE TITLE
Relational State-Space Model for Stochastic Multi-Object Systems

### ISSUE THREAD

**Issue by hkefka385** [hkefka385/paper_reading#34]
## 0. 論文 
Journal/Conference: **ICLR 2020** 
Title: **Relational State-Space Model for Stochastic Multi-Object Systems** 
Authors: Fan Yang, Ling Chen, Fan Zhou, Yusong Gao, Wei Cao
URL: https://arxiv.org/abs/2001.04050
## 1. どんなもの？ 
State-space問題に対して、状態遷移の不確実性を表す潜在変数を導入した動的な階層型潜在State-space model(SSM)を提案
状態空間の関係を柔軟に捉えるためにGNNを導入
## 2. 先行研究と比べてどこがすごい？
・状態間の関係を捉えるためにGNNを、状態遷移の不確実性を表現するために階層型潜在変数を導入 
・グラフ正規化フロー (GNF)を導入
## 4. どうやって有効だと検証した？ 
シミュレーションデータとBasketball Gameplay，Road Trafficのデータを用いて検証
## 5. 議論はある？ 
まだ十分に式や背景知識を追えてない 
(地域 + 時系列の予測問題を解くことがあればサーベイが必要になる可能性)
## 6.次に読むべき論文は？
Neural SSMについて 
Marco Fraccaro.Deep Latent Variable Models for Sequential Data. PhD thesis, 2018
Marco Fraccaro, Simon Kamronn, Ulrich Paquet, and Ole Winther.  A disentangled recognitionand nonlinear dynamics model for unsupervised learning.  InAdvances in Neural InformationProcessing Systems 30, 2017

Normalizing Flowsについて
Danilo Rezende and Shakir Mohamed. Variational inference with normalizing flows. InProceedingsof the International Conference on Machine Learning, 2015

## メモ
GNNを用いた，複数の物体の遷移シミュレーションモデルR-SSM (Relational State-space Model)を紹介
GNNを用いることで複数オブジェクトの関係を取り入れる

1 Introduction
State-space model (SSM)：状態遷移モデル
Deep SSMの導入：Marco Fraccaro.Deep Latent Variable Models for Sequential Data. PhD thesis, 2018
GNNがSSMにおいて有用だと考えられているが，決定論的な状態遷移や観測分布の過度の単純化がボトルネックと考えられている？
Breaking the softmax bottle-neck: a high-rank RNN language model. InInternational Conference on Learning Representations,2018
Marco Fraccaro, Søren Kaae Sø nderby, Ulrich Paquet, and Ole Winther. Sequential neural modelswith stochastic layers. InAdvances in Neural Information Processing Systems 29, 2016

本研究の貢献
1 GNNで物体の確率的な状態遷移をシュミレートする階層型深層SSMの提案 R-SSM：状態遷移の不確実性を表す潜在変数の導入
2 グラフ正規化フロー (GNF)を用いて表現力が豊かな分布を構築？
3 変分推論を用いてR-SSMの学習するためにstructured posterior approximationを提案 (3.2 のProposal Designに相当)

２PRELIMINARIES
V: Nノードの属性，E: エッジの情報
2.1  Graph Neural networks
グラフの頂点のラベルを再ラベル付け：Haggai Maron, Heli Ben-Hamu, Nadav Shamir, and Yaron Lipman. Invariant and equivariant graphnetworks. InInternational Conference on Learning Representations, 2019.

Multi-head attentionを用いてGNNの伝搬と頂点の計算 (e.q.1, e.q.2) / 頂点の更新もAttentionを用いて
Petar Veliˇckovi ́c, Guillem Cucurull, Arantxa Casanova, Adriana Romero, Pietro Liò, and Yoshua Bengio.  Graph attention networks.  InInternational Conference on Learning Representations,2018

2.2 State-Space Models
RNN的な状態遷移モデル

2.3 Normalizing Flows
Normalizing flows: Danilo Rezende and Shakir Mohamed. Variational inference with normalizing flows. InProceedingsof the International Conference on Machine Learning, 2015
単純な確率変数を複素数に変換？することで効率的に複雑な分布を生成可能？
https://www.slideshare.net/AkihiroNitta1/normalizing-flow-155464630
https://qiita.com/opeco17/items/62192b4dd1cd9cbaa170
→ Graph Normalizing flowsに応用

3 Relational State-Space Model
3.1 Generative Model
状態遷移モデルを時間的依存性を考慮して分解 (e.q.4)
RNNを入力として受け取るGNNを用いて観察値が生成される  (*READOUT：全ての頂点を集約してGlobal COntext vectorを作成するための関数)

3.2 Learning and Inference
Variational Sequential Monte Carlo (VSMC)の適用：対数尤度の変分下限を最大化する代わりに推論タスクの提案分布を同時に学習
Chris J Maddison, John Lawson, George Tucker, Nicolas Heess, Mohammad Norouzi, Andriy Mnih,Arnaud Doucet, and Yee Teh. Filtering variational objectives. InAdvances in Neural InformationProcessing Systems 30. 2017
C. A. Naesseth, F. Lindsten, and T. B. Schön.  High-dimensional filtering using nested sequentialmonte carlo.IEEE Transactions on Signal Processing, 67(16):4177–4188, 2019
→ Z_tのProposal design

3.3.
3.4
skip

4
GNN with sequential latent variable model (sequential LVMs)
★Deep LVMs for sequential data
RNNに確率的潜在変の組み込み
Nan Rosemary Ke, Amanpreet Singh, Ahmed Touati, Anirudh Goyal, Yoshua Bengio, Devi Parikh,and Dhruv Batra.   Modeling the long term future in model-based reinforcement learning.   InInternational Conference on Learning Representations, 2019
Anirudhand Goyal, Alessandro Sordoni, Marc-Alexandre Côté, Nan Rosemary Ke, and YoshuaBengio.  Z-forcing: Training stochastic recurrent networks.  InAdvances in Neural InformationProcessing Systems 30, 2017
Justin Bayer and Christian Osendorfer.   Learning stochastic recurrent networks.arXiv preprintarXiv:1411.7610, 2014.

遷移分布をパラメータ化しDeep SSMの開発
Rahul G Krishnan, Uri Shalit, and David Sontag. Structured inference networks for nonlinear statespace models. InAAAI Conference on Artificial Intelligence, 2017
Marco Fraccaro, Simon Kamronn, Ulrich Paquet, and Ole Winther.  A disentangled recognitionand nonlinear dynamics model for unsupervised learning.  InAdvances in Neural InformationProcessing Systems 30, 2017
Danijar Hafner, Timothy Lillicrap, Ian Fischer, Ruben Villegas, David Ha, Honglak Lee, and JamesDavidson. Learning latent dynamics for planning from pixels. InProceedings of the InternationalConference on Machine Learning, 2019

5 Experiments
5.1 Synthetic toy dataset
Table1が結果
LL: 平均対数尤度,  
MSE: 最小二条誤差
CP: 90%の予測区間のカバレッジ率
比較手法の中では優秀：GNN-AR
Thomas Kipf, Ethan Fetaya, Kuan-Chieh Wang, Max Welling, and Richard Zemel. Neural relationalinference for interacting systems.  InProceedings of the International Conference on MachineLearning, 2018.

5.2 BaksetBall GamePlay
プレイヤーとボールの関係
比較手法：Eric Zhan, Stephan Zheng, Yisong Yue, Long Sha, and Patrick Lucey.   Generating multi-agenttrajectories using programmatic weak supervision.   InInternational Conference on LearningRepresentations, 2019

5.3 Read Traffic
Table2, Table3

https://openreview.net/forum?id=B1lGU64tDr
